lib/azure/armrest/storage_account_service.rb

```diff
@@ -76,79 +76,120 @@ def list(group = armrest_configuration.resource_group)
       #   Optional. Set to 'nameAvailability' to indicate that the account
       #   name must be checked for global availability.
       #
+      # - :type
+      #   The type of storage account. The default is "Standard_GRS".
+      #
       # - :location
       #   Required: One of the Azure geo regions, e.g. 'West US'.
       #
       # - :tags
       #   A hash of tags to describe the resource. You may have a maximum of
       #   10 tags, and each key has a max size of 128 characters, and each
-      #   value has a max size of 256 characters.
-      #
-      # -:properties
-      #   - :account_type
-      #   - :custom_domains
-      #     - :custom_domain
-      #       - :name
-      #       - :use_subdomain_name
-      #--
-      # PUT
-      #
-      def create(option = {})
-        #account_name = options.fetch(:account_name)
-        #location = options.fetch(:location)
-        validating = options[:validating]
-        #tags = options[:tags]
-
-        url = @uri + "/#{account_name}"
-
-        if validating
-          url += "?validating=nameAvailability"
-        end
+      #   value has a max size of 256 characters. These are optional.
+      #
+      # Example:
+      #
+      #   sas = Azure::Armrest::StorageAccountService(config)
+      #
+      #   sas.create(
+      #     :name     => "yourstorageaccount1",
+      #     :location => "West US",
+      #     :type     => "Standard_ZRS",
+      #     :tags     => {:YourCompany => true}
+      #   )
+      #
+      # For convenience you may also specify the :resource_group as an option.
+      #
+      def create(options = {}, rgroup = armrest_configuration.resource_group)
+        rgroup ||= options[:resource_group]
+        raise ArgumentError, "No resource group specified" if rgroup.nil?
+
+        # Mandatory options
+        name = options.fetch(:name)
+        location = options.fetch(:location)
+
+        # Optional
+        tags = options[:tags]
+        type = options[:type] || "Standard_GRS"
+
+        properties = {:accountType => type}
+
+        validate_account_type(type)
+        validate_account_name(name)
+
+        url = build_url(rgroup, name)
+        url << "&validating=" << options[:validating] if options[:validating]
+
+        body = {
+          :name       => name,
+          :location   => location,
+          :tags       => tags,
+          :properties => properties
+        }.to_json
 
-        url
+        response = rest_put(url, body)
+        response.return!
```

Comment:
Basically the user isn't expecting a return object for any non-get type requests because almost none of the underlying PUT/DELETE/POST calls return a body. The caller knows creation succeeded because it didn't raise an error.